vertigis/vertigis-web-sdk
# chore: Update project to ESM and narrow dependencies

This PR performs some housekeeping. There is no new functionality for consumers. It:
 - Updates many dependencies to the latest versions
 - Converts the SDK to `"type": "module"`, i.e. uses ES modules instead of common JS. Many dependencies are now ESM only in their latest versions, making it hard to stay up to date with cjs code.
 - Pins all dev dependencies, and narrows public dependency version ranges to patch versions only. In many cases, a minor version release of e.g. TypeScript, eslint, or webpack can introduce compile/lint errors in code that previously worked. We want to make sure that consumers are using versions of these packages that we have actually tested.
 - Adds a package lock to facilitate npm audit.
 - Adds checks for minimum versions of Node and npm that we support. We use modern JS syntax that will not work in Node 12 or earlier.
 - Adds ease of life VS Code settings for maintainers.
 
## Testing Suggestions
- Make sure that the various scripts (create, start, build, test) work from the SDK repo itself.
- Make sure that the consumer project scripts (start, build, upgrade) work from both a new project created with this build of the SDK, and from an existing project created with an older version of the SDK that is upgraded to use this build.

For the latter, `npm link` won't work very well. I recommend using `npm pack` to create a *.tgz file from this branch, then `npm install <path-to-sdk-tgz>` from within the library project that is being used for testing.
